Sources/PitchSpeller/Data Structures/FlowNetwork.swift

```diff
@@ -41,9 +41,7 @@ public struct FlowNetwork <Node: Hashable> {
         }
             
         func pushFlow (through path: UnweightedGraph<Node>.Path) {
-            let minimumEdge = (path.adjacents.compactMap {
-                residualNetwork.weight($0)
-                }.min())!
+            let minimumEdge = (path.adjacents.compactMap(residualNetwork.weight).min())!
```

Comment:
Does this whole thing need to be wrapped in parens, or can the forced optional unwrapping happen only at the `min()` call, e.g.:

```Swift
let minimumEdge = path.adjacents.compactMap(residualNetwork.weight).min()!
```

As the compact map will return an `[Weight]` (non-optional), where the `min()` call will return `nil` if this array is empty.

Which brings up an interesting question: do we have guarantees somewhere that a `Path` (and therefore `path.adjacents`) will always be non-empty? That could be worth looking into.